x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/score/Decay.java

```diff
@@ -234,6 +237,8 @@ private TypeResolution validateOriginAndScale(DataType valueType) {
             );
         } else if (isMillisOrNanos(valueType)) {
             return validateOriginAndScale(DataType::isMillisOrNanos, "datetime or date_nanos", DataType::isTimeDuration, "time_duration");
+        } else if (isUnsignedLong(valueType)) {
+            return validateOriginAndScale(DataType::isUnsignedLong, "unsigned long", DataType::isUnsignedLong, "unsigned_long");
```

Comment:
I wonder if it's enough to check if origin and scale are numeric? It should be possible to mix and match numeric types, i.e. use an unsigned long origin but an int scale etc

Reply:
That's a good point, I wanted to restrict it first to avoid "wrong" calculations, but giving it a second thought it definitely makes sense to support mixed numeric types.

Adjusted with:
- [Support mixed numeric types for numeric decay](https://github.com/elastic/elasticsearch/pull/134440/commits/f43b505dfe56184723bc694da7aec3ccd60513e5)
- [Add mixed numeric types test cases](https://github.com/elastic/elasticsearch/pull/134440/commits/13199036825f88747bb53a6d4c842aea7a5f68b0)

Full disclosure: we need to the same for `offset`. I'll add it to https://github.com/elastic/elasticsearch/issues/134789, which I've opened yesterday and address it in a separate PR.